src/flows/pipes/MetricSelector/view.tsx

```diff
@@ -1,33 +1,48 @@
 // Libraries
-import React, {FC} from 'react'
+import React, {FC, useContext} from 'react'
 // Types
 import {PipeProp} from 'src/types/flows'
+import {Bucket} from 'src/types'
 
 // Contexts
 import BucketProvider from 'src/flows/context/buckets'
 import {SchemaProvider} from 'src/flows/pipes/MetricSelector/context'
+import {PipeContext} from 'src/flows/context/pipe'
 
 // Components
 import BucketSelector from 'src/flows/shared/BucketSelector'
 import FieldsList from 'src/flows/pipes/MetricSelector/FieldsList'
 import FilterTags from 'src/flows/pipes/MetricSelector/FilterTags'
 import SearchBar from 'src/flows/pipes/MetricSelector/SearchBar'
 
-const DataSource: FC<PipeProp> = ({Context}) => (
-  <BucketProvider>
-    <SchemaProvider>
-      <Context>
-        <div className="data-source--controls">
-          <div className="data-source--bucket">
-            <BucketSelector />
+const DataSource: FC<PipeProp> = ({Context}) => {
+  const {data, update} = useContext(PipeContext)
+  const updateBucket = (bucket: Bucket) => {
+    update({
+      bucket,
+    })
+  }
+
+  return (
+    <BucketProvider>
+      <SchemaProvider>
+        <Context>
+          <div className="data-source--controls">
+            <div className="data-source--bucket">
+              <BucketSelector
+                selected={data.bucket}
+                onSelect={updateBucket}
+                style={{width: '250px', flex: '0 0 250px'}}
```

Comment:
Why like this?

Reply:
because you did this?
https://github.com/influxdata/ui/pull/1154/files/8ef250f7532877c8ed1c17bbf7be8d43599128d5#diff-c13c275699d8fc942664ebe6c5b6b9c8a1d23b2fd88a25f15bac558430172815L99

you see.. you go over there... and you copy... then you go over here.... and you pasta

Reply:
My garbage is expected, I have higher standards for you 🙂

Reply:
:rofl: